services/data/src/react/hooks/useQueryExecutor.ts

```diff
@@ -10,6 +10,14 @@ interface StateType<T> {
     data?: T
 }
 
+const checkVariables = (variables: any = {}) => {
+    if (typeof variables !== 'object' || Array.isArray(variables)) {
```

Comment:
Why did you add the array check? If the value is an array, `typeof variables` should be `array` and therefore should be caught by `typeof variables !== 'object'` already

Reply:
`typeof [] === 'object'` :eyes:

Reply:
JavaScript is really weird in that arrays are actually just objects (`Array.from({ 0: 'a', length: 1 }) => ['a']`) and so `typeof` of an array is `"object"`. See http://web.mit.edu/jwalden/www/isArray.html

Reply:
That's weird.. I was sure that it's always `array`.. nvm then